examples/extended/parallel/MPI/source/src/G4MPImanager.cc

```diff
@@ -84,7 +85,10 @@ G4MPImanager::G4MPImanager(int nof_extra_workers)
     nof_extra_workers_(nof_extra_workers)
 {
   // MPI::Init();
-  MPI::Init_thread(MPI::THREAD_SERIALIZED);
+  MPI_Init_thread(nullptr, nullptr, MPI_THREAD_SERIALIZED, &provided_);
+  if (provided_ < MPI_THREAD_SERIALIZED) {
+    std::cerr << "Warning: MPI does not provide the requested threading support!" << std::endl;
```

Comment:
```suggestion
    G4cout << "Warning: MPI does not provide the requested threading support!" << G4endl;
```

What's the behaviour if this condition is true? Will the application still run, or is it a hard error condition?

Reply:
I'm not too sure, I will need to test it out

Reply:
If you could, that'd be great to clarify this.

Reply:
I have been trying to trigger this condition on my machine, but without actual success. I have only been able to trigger this replacing provided_ with MPI_THREAD_SINGLE.

In theory, the application should not run if there are problems with the MPI threading level, so it should be left as such. We definitely don't want anything below SERIALISED. As least, this is the strong suggestion that the documentation makes: https://docs.open-mpi.org/en/v5.0.x/man-openmpi/man3/MPI_Init_thread.3.html#mpi-init-thread

Reply:
Thanks for investigating! Given that running with a level below `SERIALIZED` isn't good, this should be reported as a fatal `G4Exception` to terminate the application at this point. Something like:

```c++
int provided_;
MPI_Init_thread(nullptr, nullptr, MPI_THREAD_SERIALIZED, &provided_);
if(provided_ < MPI_THREAD_SERIALIZED)
{
  G4Exception("G4MPImanager::G4MPImanager()",
              "G4MPImanager001",
               FatalException,
	       "MPI Initialization failed to setup with MPI_THREAD_SERIALIZED or better");
}
```

Reply:
Good idea, fixed!